docs/articles/documentation/guides/advanced-guides/test-https-features-and-http2-websites.md

```diff
@@ -23,21 +25,95 @@ If the tested page does not use HTTPS-specific features (like
 [Service Workers](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API),
 [Geolocation API](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation_API),
 [ApplePaySession](https://developer.apple.com/documentation/apple_pay_on_the_web/applepaysession), or
-[SubtleCrypto](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto)), TestCafe successfully proxies it over HTTP. Otherwise, the tests fail because the page throws JavaScript errors, does not load or loads partially. This indicates that you should [configure the proxy server to use HTTPS](#proxy-the-website-over-https).
+[SubtleCrypto](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto)), TestCafe successfully proxies it over HTTP. Otherwise, the tests fail because the page throws JavaScript errors, does not load or loads partially.
 
-### Proxy the Website Over HTTPS
+### Use a Self-Signed Certificate
+
+To establish a secure HTTPS-connection, Node.js requires a valid [PKCS12](https://en.wikipedia.org/wiki/PKCS_12) certificate (`.pfx`). To launch tests from the command line, you need to [generate one manually](#generate-a-certificate-from-the-command-line). In API, use an [automatic module](#generate-a-certificate-in-your-code), unless you are willing to [use a trusted certificate](#use-a-trusted-certificate).
```

Comment:
> To launch tests from the command line, you need to [generate one manually](#generate-a-certificate-from-the-command-line).

This sentence repeats the structure of the previous sentence. This is not always bad, it can even be intended, but here sentences 2 and 3 detail the first sentence, so it's not quite apt.

Secondly, "one" in "generate one manually" has no obvious connection to "certificate" to me, because it's a different sentence. "It" is better.

And since we got rid of "automatic" and "manual" in the section titles, let's remove them from here as well.

And "unless" should oppose both sentences 2 and 3, shouldn't it? Because you needn't use SSL or Node.js modules if you have a trusted certificate.

Which brings us to version 2:

`To establish a secure HTTPS-connection, Node.js requires a valid [PKCS12](https://en.wikipedia.org/wiki/PKCS_12) certificate (`.pfx`). [Generate it with `openssl`](#generate-a-certificate-from-the-command-line) if you launch tests from the command line. In the API, you can use a [Node.js module to generate pfx](#generate-a-certificate-in-your-code). Alternatively, you can [use a trusted certificate](#use-a-trusted-certificate).`

That's okay, but shouldn't we say about trusted certificates somewhere above the "Self-signed certificate section"? Let's try:

```
...Otherwise, the tests fail because the page throws JavaScript errors, does not load or loads partially.

To establish a secure HTTPS-connection, Node.js requires a valid [PKCS12](https://en.wikipedia.org/wiki/PKCS_12) certificate (`.pfx`). If you have a certificate signed by a trusted authority, we recommend that you [use this certificate](#use-a-trusted-certificate) if your security policy permits. The browsers are then able to open the tested page without workarounds. Alternatively, you can [generate and use a self-signed certificate](#use-a-self-signed-certificate).

### Use a Self-Signed Certificate

[Generate a self-signed certificate with `openssl`](#generate-a-certificate-from-the-command-line) if you launch tests from the command line. In the API, you can use the [openssl-self-signed-certificate](https://www.npmjs.com/package/openssl-self-signed-certificate) module to [generate a certificate in code](#generate-a-certificate-in-your-code). 
```